### ISSUE TITLE
glooctl istio inject causes gateway proxy failure

### ISSUE THREAD

**Issue by jameshbarton** [solo-io/gloo#3650]
**Describe the bug**
When using the `glooctl istio inject` command with Gloo open-souce, it causes the gateway-proxy pod to fail and fall into a CrashLoopBackOff state.

Note that this does not appear to be directly related to recently closed #3641 .  I confirmed that `sds` is using the correct 1.5.0-beta25 version.
```
% glooctl version
Client: {"version":"1.5.0-beta25"}
Server: {"type":"Gateway","kubernetes":{"containers":[{"Tag":"1.5.0-beta25","Name":"discovery","Registry":"quay.io/solo-io"},{"Tag":"1.5.0-beta25","Name":"gateway","Registry":"quay.io/solo-io"},{"Tag":"1.5.0-beta25","Name":"gloo-envoy-wrapper","Registry":"quay.io/solo-io"},{"Tag":"1.5.0-beta25","Name":"sds","Registry":"quay.io/solo-io"},{"Tag":"1.7.1","Name":"proxyv2","Registry":"docker.io/istio"},{"Tag":"1.5.0-beta25","Name":"gloo","Registry":"quay.io/solo-io"}],"namespace":"gloo-system"}}
```

**To Reproduce**
Following [this guide](https://docs.solo.io/gloo/latest/guides/integrations/service_mesh/gloo_istio_mtls/) but with Istio 1.7.1.
1. Install Istio 1.7.1
```
curl -L https://istio.io/downloadIstio | ISTIO_VERSION=1.7.1 sh -
cd istio-1.7.1
istioctl install --set profile=demo
```
2. Install bookinfo sample app.
```
kubectl label namespace default istio-injection=enabled
kubectl apply -f samples/bookinfo/platform/kube/bookinfo.yaml
```
3. Install Gloo 1.5.0-beta25
```
kubectl create ns gloo-system; helm install --namespace gloo-system --version 1.5.0-beta25 gloo gloo/gloo
```
4. Add the Istio mTLS secret to the gateway-proxy pod with `glooctl istio inject`.
5. Observe gateway-proxy in CrashLoopBackOff state.  See also the log messages in second screenshot.

<img width="1250" alt="Screen Shot 2020-09-21 at 10 24 14 AM" src="https://user-images.githubusercontent.com/52334260/93779197-b37b0f80-fbf4-11ea-865c-8e88399ac438.png">

<img width="1259" alt="Screen Shot 2020-09-21 at 9 38 59 AM" src="https://user-images.githubusercontent.com/52334260/93779221-baa21d80-fbf4-11ea-8d6a-25a394f59929.png">

**Expected behavior**
Gateway Proxy continues to work as expected with Istio mTLS secret injected.

**Additional context**
- Gloo version 1.5.0-beta25
- Kubernetes version 1.16.3-gke.1
- Istio version 1.7.1

**Comment by Sodman** [comment]
I was able to reproduce this locally with Istio 1.7.1. This issue appears specific to Istio 1.7.1, this still works fine on Istio 1.7.0.

If this is something we can fix with the sidecar itself, assuming it's backwards compatible to 1.7.0, we can update [the existing 1.7 sidecar code](https://github.com/solo-io/gloo/blob/master/projects/gloo/cli/pkg/cmd/istio/sidecars/istio1.7.go#L8). If the fix is not backwards compatible, we may need a separate 1.7.1+ container 😞.

**Comment by christian-posta** [comment]
@jameshbarton @Sodman we chatted about this on slack.. we have a fix for this. I'll let Shane run with it.

**Comment by jameshbarton** [comment]
Confirmed that this problem is resolved using GlooE v1.5.0, Istio v1.7.3, and Kubernetes v1.16.13-gke.1
